src/index/ivf_flat.rs

```diff
@@ -0,0 +1,234 @@
+//! Interface and implementation to IVFFlat index type.
+
+use super::*;
+
+use crate::error::Result;
+use crate::faiss_try;
+use std::mem;
+use std::ptr;
+
+/// Alias for the native implementation of a flat index.
+pub type IVFFlatIndex = IVFFlatIndexImpl;
+
+/// Native implementation of a flat index.
+#[derive(Debug)]
+pub struct IVFFlatIndexImpl {
+    inner: *mut FaissIndexIVFFlat,
+}
+
+unsafe impl Send for IVFFlatIndexImpl {}
+unsafe impl Sync for IVFFlatIndexImpl {}
+
+impl CpuIndex for IVFFlatIndexImpl {}
+
+impl Drop for IVFFlatIndexImpl {
+    fn drop(&mut self) {
+        unsafe {
+            faiss_IndexIVFFlat_free(self.inner);
+        }
+    }
+}
+
+impl IVFFlatIndexImpl {
+    /// Create a new IVF flat index.
+    pub fn new(quantizer: flat::FlatIndex, d: u32, nlist: u32, metric: MetricType) -> Result<Self> {
+        unsafe {
+            let metric = metric as c_uint;
+            let mut inner = ptr::null_mut();
+            faiss_try(faiss_IndexIVFFlat_new_with_metric(
+                &mut inner,
+                quantizer.inner_ptr(),
+                d as usize,
+                nlist as usize,
+                metric,
+            ))?;
+
+            mem::forget(quantizer); // own_fields default == true
```

Comment:
leak is here
IVF inherited from `Level1Quantizer`
https://github.com/facebookresearch/faiss/blob/master/faiss/IndexIVF.h#L95

`own_fields` default == false https://github.com/facebookresearch/faiss/blob/master/faiss/IndexIVF.cpp#L40